### PR TITLE
NODE-319 Node should not request an extension before a previous one is applied

### PR DESCRIPTION
**Pull request by vsuharnikov**
* A node doesn't break an extension loading when mining;
* A node doesn't mine a microblock during a block mining.

### CODE REVIEW

File: src/main/scala/com/wavesplatform/network/messages.scala [hunk 1]
```diff
@@ -36,4 +36,18 @@ object MicroBlockInv{
...
+
+object LocalScoreChanged {
+  sealed trait Reason extends Product with Serializable
+  object Reason {
+    val All: Set[Reason] = Set(ForkApplied, Rollback, Checkpoint, Other)
+
+    case object ForkApplied extends Reason
+    case object Rollback extends Reason
+    case object Checkpoint extends Reason
+    case object Other extends Reason
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/network/messages.scala]: Other = BlockApplied?

File: src/main/scala/com/wavesplatform/network/CoordinatorHandler.scala [hunk 2]
```diff
@@ -43,9 +43,9 @@ class CoordinatorHandler(checkpointService: CheckpointService,
   private val processBlock = Coordinator.processSingleBlock(checkpointService, history, blockchainUpdater, time, stateReader, utxStorage, settings.blockchainSettings, featureProvider) _
   private val processMicroBlock = Coordinator.processMicroBlock(checkpointService, history, blockchainUpdater, utxStorage) _
 
-  private def scheduleMiningAndBroadcastScore(score: BigInt): Unit = {
+  private def scheduleMiningAndBroadcastScore(reason: LocalScoreChanged.Reason = LocalScoreChanged.Reason.Other)(score: BigInt): Unit = {
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/network/CoordinatorHandler.scala]: What's the point of using  default value?

Comment (vsuharnikov) [reply src/main/scala/com/wavesplatform/network/CoordinatorHandler.scala]: To not break extension loading

File: src/main/scala/com/wavesplatform/network/messages.scala (hunk 1 above)
Comment (vsuharnikov) [reply src/main/scala/com/wavesplatform/network/messages.scala]: Yes, but I'll change it to flag

**Pull request merged**